### PR TITLE
Add pysoundcard

### PR DESCRIPTION
**Pull request by jan-janssen**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jan-janssen** [comment]
```
import: 'soundcard'
Traceback (most recent call last):
  File "/home/conda/staged-recipes/build_artifacts/pysoundcard_1717834548449/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/pypy3.9/site-packages/soundcard/pulseaudio.py", line 17, in <module>
    _pa = _ffi.dlopen('pulse')
  File "/home/conda/staged-recipes/build_artifacts/pysoundcard_1717834548449/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/pypy3.9/cffi/api.py", line 150, in dlopen
    lib, function_cache = _make_ffi_library(self, name, flags)
  File "/home/conda/staged-recipes/build_artifacts/pysoundcard_1717834548449/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/pypy3.9/cffi/api.py", line 832, in _make_ffi_library
    backendlib = _load_backend_lib(backend, libname, flags)
  File "/home/conda/staged-recipes/build_artifacts/pysoundcard_1717834548449/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/pypy3.9/cffi/api.py", line 827, in _load_backend_lib
    raise OSError(msg)
OSError: ctypes.util.find_library() did not manage to locate a library called 'pulse'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/conda/staged-recipes/build_artifacts/pysoundcard_1717834548449/test_tmp/run_test.py", line 2, in <module>
    import soundcard
  File "/home/conda/staged-recipes/build_artifacts/pysoundcard_1717834548449/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/pypy3.9/site-packages/soundcard/__init__.py", line 4, in <module>
    from soundcard.pulseaudio import *
  File "/home/conda/staged-recipes/build_artifacts/pysoundcard_1717834548449/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/pypy3.9/site-packages/soundcard/pulseaudio.py", line 20, in <module>
    _pa = _ffi.dlopen('libpulse.so')
  File "/home/conda/staged-recipes/build_artifacts/pysoundcard_1717834548449/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/pypy3.9/cffi/api.py", line 150, in dlopen
    lib, function_cache = _make_ffi_library(self, name, flags)
  File "/home/conda/staged-recipes/build_artifacts/pysoundcard_1717834548449/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/pypy3.9/cffi/api.py", line 832, in _make_ffi_library
    backendlib = _load_backend_lib(backend, libname, flags)
  File "/home/conda/staged-recipes/build_artifacts/pysoundcard_1717834548449/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/pypy3.9/cffi/api.py", line 827, in _load_backend_lib
    raise OSError(msg)
OSError: Cannot load library libpulse.so: libpulse.so: cannot open shared object file: No such file or directory.  Additionally, ctypes.util.find_library() did not manage to locate a library called 'libpulse.so'
```

**Comment by jan-janssen** [comment]
This package could be `noarch` but on `linux` it is required to have `pulseaudio` installed. Is it possible to create an `noarch` package with operation system specific dependencies? To me this sounds like a requirement for operation system specific builds even though nothing is compiled during the build process.

**Comment by jan-janssen** [comment]
@conda-forge/staged-recipes, ready for review

**Comment by jan-janssen** [comment]
@conda-forge/help-python

File: recipes/pysoundcard/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+requirements:
+  host:
+    - python
+    - pip
+  run:
+    - python
+    - pulseaudio  # [linux]
```
Comment (ocefpaf) [inline recipes/pysoundcard/meta.yaml:22]: If this is optional and you still get a functional Windows and macOS package without it you can use the OS-noarch: https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-packages-with-os-specific-dependencies

**Pull request merged**